src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/RSAKeyPemTests.cs

```diff
@@ -486,8 +486,8 @@ public static void ImportFromEncryptedPem_NoEncryptedPem()
         {
             using (RSA rsa = RSAFactory.Create())
             {
-                string pem = @"
------BEGIN PRIVATE KEY-----
+                string pem = $@"
+-----{TestSecretHelper.BEGIN_PRIVATE_KEY}-----
```

Comment:
How does this fix the issue?  Is this effectively just making the code more convoluted such that the current tooling gets confused and doesn't see this as an issue?  If so, that doesn't seem like the right fix.

Reply:
> Is this effectively just making the code more convoluted such that the current tooling gets confused and doesn't see this as an issue?

Yes

> If so, that doesn't seem like the right fix.

I would like to use `// [SuppressMessage]` everywhere which would be consistent across the codebase. This mangling is result of our team discussion. I'm open to change this to whatever we agree.

The only benefit of this solution is that the code is still readable and it explicitly mark the key as know hardcoded value. The purpose of the TestSecretHelper class is just to annotate known test credential

Reply:
I'd rather use the SuppressMessage.  It's much clearer with its intent.

Reply:
> I'd rather use the SuppressMessage. It's much clearer with its intent.

+1